datasets/pec/pec.py

```diff
@@ -0,0 +1,205 @@
+        # TODO: Downloads the data and defines the splits
+        # dl_manager is a datasets.download.DownloadManager that can be used to
+        # download and extract URLs
+        dl_dir = dl_manager.download_and_extract(_URL)
+        data_dir = os.path.join(dl_dir, "hf_pec")
+        print("The downloaded data has been saved to ", data_dir)
+        if self.config.domain in ["happy", "offmychest"]:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "train.txt"), "split": "train"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "test.txt"), "split": "test"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={"filepath": os.path.join(data_dir, self.config.domain, "valid.txt"), "split": "dev"},
+                ),
+            ]
+        else:
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "train.txt"), "split": "train"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.TEST,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "test.txt"), "split": "test"},
+                ),
+                datasets.SplitGenerator(
+                    name=datasets.Split.VALIDATION,
+                    gen_kwargs={"filepath": os.path.join(data_dir, "valid.txt"), "split": "dev"},
+                ),
+            ]
+
+    def _generate_examples(self, filepath, split):
+        """ Yields examples. """
+        # TODO: Yields (key, example) tuples from the dataset
+        context_speakers = []
+        context = []
+        response_speaker = ""
+        response = ""
+        example_id = 0
+        data_dir = os.path.dirname(filepath)
+
+        if self.config.domain == "all":
+            # load files from both happy and offmychest
+            filepath = [
+                os.path.join(data_dir, "happy", os.path.basename(filepath)),
+                os.path.join(data_dir, "offmychest", os.path.basename(filepath)),
+            ]
+        else:
+            filepath = [filepath]
+
+        # create persona
+        if not hasattr(self, "persona_" + self.config.domain):
+            persona_paths = [os.path.join(os.path.dirname(fpath), "persona.txt") for fpath in filepath]
+            setattr(self, "persona_" + self.config.domain, self._load_persona(persona_paths))
+
+        persona = getattr(self, "persona_" + self.config.domain)
+        # print("Domain {0}: number of unique personas: {1}".format(self.config.domain, len(persona)))

```

Comment:
Why not simply use this instead : 
```
persona_paths = [os.path.join(os.path.dirname(fpath), "persona.txt") for fpath in filepath]
persona = self._load_persona(persona_paths)
```
?

(or maybe I'm missing something)

Reply:
I dont want to re-create the `persona` for every split of the dataset since it is shared by all splits.

Reply:
Then feel free to generate the persona once in the `_split_generator` method (it's called once for all splits), and pass the persona object through the `gen_kwargs` that are passed to `_generate_examples `
This way the persona will be created only once and reused in all the splits.
Does that make sense ?